### ISSUE TITLE
FIT_ROUND RECEIVED 0 RESULTS 10 FAILURES SIMULATION PYTORCH

### ISSUE THREAD

**Issue by Divyanshupy** [adap/flower#1157]
### Describe the bug

Hello, I am trying to port the simulation PyTorch example to PyTorch Lightning but I am getting fit_round received 0 results error. However, when using the same code without simulation in a client server setting is working without any error. I debugged the client's fit function and it is working as expected. 

Thanks in advance.

### Steps/Code to Reproduce

class my_client(fl.client.NumPyClient):

    def __init__(self, cid,fed_path):
        self.cid = cid
        self.fed_dir = fed_path
        self.model = fdc.model(1e-04,0.01)
        self.properties = {"tensor_type": "numpy.ndarray"}
        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
    def get_parameters(self):
        return [v.cpu().numpy() for k, v in self.model.state_dict().items()]
            
    def set_parameters(self, parameters):
        set_weights(self.model, parameters)
    
    def get_properties(self, ins):
        return self.properties
 
    
    def fit(self, parameters, config):
        print(config)
        self.set_parameters(parameters)
        num_workers = 32
        # len(ray.worker.get_resource_ids()["CPU"])
        trainloader = get_dataloader(
            self.fed_dir,
            self.cid,
            is_train=True,
            batch_size=64,
            workers=num_workers)
        
        trainer = pl.Trainer(max_epochs=10, gpus=[3])
        trainer.fit(self.model,trainloader)
        # results = trainer.validate(self.model,valloader)
        # precision = results[0]['val_precision']
        # recall    = results[0]['val_recall']
        # accuracy  = results[0]['val_accuracy']
        #wandb.log({'val_precision':precision,'val_recall':recall,'val_accuracy':accuracy})
        return self.get_parameters(), len(trainloader.dataset), {}
    
    def evaluate(self, parameters, config):
        return super().evaluate(parameters, config)
    
#GLOB funcs
def set_weights(model: torch.nn.ModuleList, weights: fl.common.Weights) -> None:
    """Set model weights from a list of NumPy ndarrays."""
    state_dict = OrderedDict(
        {
            k: torch.tensor(np.atleast_1d(v))
            for k, v in zip(model.state_dict().keys(), weights)
        }
    )
    model.load_state_dict(state_dict, strict=True)

def fit_config(rnd: int) -> Dict[str, str]:
    """Return a configuration with static batch size and (local) epochs."""
    config = {
        "epoch_global": str(rnd),
        "epochs": str(10),  # number of local epochs
        "batch_size": str(64),
    }
    return config

def get_eval_fn(testset):
    
    def evaluate(weights):
        model = fdc.model(1e-04,0.01).eval()
        set_weights(model,weights)  # Update model with the latest parameters
        test_trainer = pl.Trainer(max_epochs=1,gpus=[1])
        test_dl = torch.utils.data.DataLoader(testset, batch_size=64)
        dic = test_trainer.test(model,test_dl)
        loss = dic[0]['test_loss_epoch']
        precision = dic[0]['test_precision_epoch']
        recall = dic[0]['test_recall_epoch']
        accuracy = dic[0]['test_accuracy_epoch']
        #wandb.log({'test_precision':precision,'test_recall':recall,'test_accuracy':accuracy})
        return loss,{'precision':precision,'recall':recall,'accuracy':accuracy}
        
    return evaluate

def main():
    
    pool_size = 100 # number of dataset partions (= number of total clients)
    client_resources = {
        "num_cpus": 1
    }  # each client will get allocated 1 CPUs

    # download CIFAR10 dataset
    train_path, testset = getCIFAR10()
    
    fed_dir = do_fl_partitioning(
        train_path, pool_size=pool_size, alpha=1000, num_classes=10, val_ratio=0.1
    )
    
    # configure the strategy
    strategy = fl.server.strategy.FedAvg(
        fraction_fit=0.5,
        min_fit_clients=5,
        min_available_clients=pool_size,  # All clients should be available
        on_fit_config_fn=fit_config,
        eval_fn=get_eval_fn(testset)
         # centralised testset evaluation of global model
    )
    
    def client_fn(cid: str):
        # create a single client instance
        return my_client(cid, fed_dir)

    # (optional) specify ray config
    ray_config = {"include_dashboard": False}

    # start simulation
    fl.simulation.start_simulation(
        client_fn=client_fn,
        num_clients=pool_size,
        client_resources=client_resources,
        num_rounds=10,
        strategy=strategy,
        ray_init_args=ray_config,
    )

if __name__ == "__main__":
    main()

### Expected Results

All clients should return the results.

### Actual Results

![Screenshot 2022-03-26 at 4 33 20 PM](https://user-images.githubusercontent.com/36807028/160236610-a099ad32-e9f2-462e-be81-33c3460325f4.png)
